jarviscli/plugins/teachme.py

```diff
@@ -0,0 +1,420 @@
+            testLimit = int(jarvis.input("Enter a limit of exercices. "))
+            break
+        except ValueError:
+            jarvis.say("Please enter a valid limit.")
+    if testLimit >= len(testyourself):
+        testLimit = len(testyourself)
+
+    if testCheck.strip().lower() == 'y':
+        userAsnwer = []
+        random.shuffle(testyourself)
+        if saving:
+            f.write('\t\t\t\t\t\t\t\t\t\tTest Yourself\n')
+            f.write('\n')
+        jarvis.say('\t\t\t\t\t\t\t\t\t\tTest Yourself\n')
+        jarvis.say('\n')
+        ids = 0
+        for i in testyourself:
+            ids = ids + 1
+            if ids > testLimit:
+                break
+            if saving:
+                f.write(str(ids) + '. ' + i[0] + ' (Hint: ' + i[1] + ')\n')
+            jarvis.say(str(ids) + '. ' + i[0] + ' (Hint: ' + i[1] + ')\n')
+            userAsnwer.append(input("Your answer: "))
+        if saving:
+            f.write('\n')
+            f.write('\t\t\t\t\t\t\t\t\t\tAnswer Keys\n')
+            f.write('\n')
+        jarvis.say('\n')
+        jarvis.say('\t\t\t\t\t\t\t\t\t\tAnswer Keys\n')
+        jarvis.say('\n')
+
+        ids = 0
+        correctAnswer = 0
+        for i in testyourself:
+            ids = ids + 1
+            if ids > testLimit:
+                break
+            flag = False
+            s = SequenceMatcher(None, userAsnwer[ids - 1].strip().lower(), i[2].strip().lower())
+            if s.ratio() > .8:
+                correctAnswer += 1
+                flag = True
+            if saving:
+                f.write(str(ids) + '. Your answer: ' + userAsnwer[ids - 1] + '| Correct Answer:' + i[2] + '| ' + str(
+                    flag) + '\t')
+            jarvis.say(str(ids) + '. Your answer: ' + userAsnwer[ids - 1] + '| Correct Answer:' + i[2] + '| ' + str(
+                flag) + '\t')
+        score = correctAnswer / len(testyourself)
+        if saving:
+            f.write('Your score is ' + str(np.round(score, 2)) + '\n')
+        jarvis.say('Your score is ' + str(np.round(score, 2)) + '\n')
+    if saving:
+        jarvis.say('Writing file terminated. Refer to ' + loc + filename + '.txt')
+
+
+@plugin('teach me')
+def call(jarvis, s):
+    """
+    This is an application which extracts the unique words related to a query from internet and news, then generates a .txt output file that contains the definition, etymology, example, and question about the extracted words.
+    """
+    jarvis.say(
+        'This is an application which extracts the unique words related to a query from internet and news, then generates a .txt output file that contains the definition, etymology, example, and question about the extracted words.')
+    jarvis.say(
+        'Note that you need to have an oxford api-key in order to continue the process. If you do not possess the key, please refer to https://developer.oxforddictionaries.com/ to obtain your api-key.')
+    apiKey = '0611eb618d244d15af619451f9d2007b'

```

Comment:
Guess this line shouldn't be in this PR?
Please make sure, this api key is invalid ;)

Reply:
That's a woupsy I added a prompt for the news API key ^^'